runtime/executor/src/main/java/edu/snu/nemo/runtime/executor/TaskGroupExecutor.java

```diff
@@ -60,14 +52,36 @@
   private final DataTransferFactory channelFactory;
   private final MetricCollector metricCollector;
 
-  /**
-   * Map of task IDs in this task group to their readers/writers.
-   */
-  private final Map<String, List<InputReader>> physicalTaskIdToInputReaderMap;
-  private final Map<String, List<OutputWriter>> physicalTaskIdToOutputWriterMap;
+  private final List<InputReader> inputReaders;
+  private final Map<InputReader, List<TaskDataHandler>> inputReaderToDataHandlersMap;
+  private final Map<String, Iterator> idToSrcIteratorMap;
+  private final Map<String, List<TaskDataHandler>> srcIteratorIdToDataHandlersMap;
+  private final Map<String, List<TaskDataHandler>> iteratorIdToDataHandlersMap;
+  private final LinkedBlockingQueue<Pair<String, DataUtil.IteratorWithNumBytes>> partitionQueue;
+  private List<TaskDataHandler> taskDataHandlers;
```

Comment:
Do we have to maintain this list and `srcIteratorIdToDataHandlersMap`, `srcIteratorIdToDataHandlersMap` at the same time?